src/main/controller/UserController.ts

```diff
@@ -156,15 +158,31 @@ export class UserController {
         const id = await IdValidation.schema.validateAsync(req.params.id, {
           abortEarly: false,
         });
-        const userRepository = transactionalEntityManager.getCustomRepository(
-          UserRepository
+
+        // get all ids of promotions that user has saved
+        const rawPromotionIds = await transactionalEntityManager
+          .createQueryBuilder()
+          .select('savedPromotions.promotionId')
+          .from(SavedPromotion, 'savedPromotions')
+          .where('savedPromotions.userId = :id', { id })
+          .cache(true)
+          .getRawMany();
+        const promotionIds = rawPromotionIds.map(
+          (rawPromotion) => rawPromotion.savedPromotions_promotionId
         );
-        const savedPromotions = await userRepository.findOneOrFail(id, {
-          relations: ['savedPromotions', 'savedPromotions.promotion'],
-          cache: true,
-        });
 
-        res.status(200).send(savedPromotions);
+        let promotions: Promotion[] = [];
+
+        if (promotionIds.length !== 0) {
+          // get all promotions using the promotion id's, DO NOT join to discount/schedules, we don't need all the information
+          promotions = await transactionalEntityManager
+            .getCustomRepository(PromotionRepository)
+            .find({
+              id: In(promotionIds),
+            });
```

Comment:
Changed the contract for getting a user's saved promotions. I didn't see the point of returning information about the user, so I just returned an array of promotions. I also don't think we need to join to discount or schedules, don't think that information is necessary for viewing a user's saved promotions.

Reply:
Just kidding, I think we do need to return discount or schedules as well...will fix this

Reply:
Okok I think to be safe I'll just join everything again